torchgeo/datasets/utils.py

```diff
@@ -772,6 +776,47 @@ def path_is_vsi(path: Path) -> bool:
     return '://' in str(path) or str(path).startswith('/vsi')
 
 
+def listdir_vsi_recursive(root: Path) -> list[Path]:
+    dirs = [root]
+    files = []
+    while dirs:
+        dir = dirs.pop()
+        try:
+            subdirs = fiona.listdir(dir)
+            dirs.extend([os.path.join(dir, subdir) for subdir in subdirs])
+        except FionaValueError:
```

Comment:
@adamjstewart the test [test_files_property_for_virtual_files](https://github.com/microsoft/torchgeo/blob/8c02cec3801672947455db92f76537072b15dd4a/tests/datasets/test_geo.py#L181) checks vsi paths that (currently) does not exist. If I was to raise error upon 'does not exist', the test would fail.

Three options:
1. Allow non-existing vsi to pass through without raising errors
2. Remove the test, and rely on the new test I added that checks only zip
3. Add tests for all the vsi listed in the test.

For now I went with number 1.

Reply:
Agreed. Removed the test, added new tests for files within zip archive, and made GeoDataset.files warn the same way as for non-existing (regular) directores.